### ISSUE TITLE
[Bug]: React v18 build error: Unexpected token

### ISSUE THREAD

**Issue by cihatertem** [remix-run/react-router#8827]
### What version of React Router are you using?

6.3.0

### Steps to Reproduce

{
  "name": "setup",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "axios": "^0.27.2",
    "react": "^18.1.0",
    "react-dom": "^18.1.0",
    "react-router-dom": "^6.3.0",
    "react-scripts": "0.9.5"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test --env=jsdom",
    "eject": "react-scripts eject"
  }
}

npm run build gives that error:"
static/js/main.cd81754f.js from UglifyJs
SyntaxError: Unexpected token: punc ({) [./~/react-router-dom/umd/react-router-dom.production.min.js:11,866]
"

and build fails!

### Expected Behavior

build success.

### Actual Behavior

When I run "npm run build" command, build fails with this error msg:
static/js/main.cd81754f.js from UglifyJs
SyntaxError: Unexpected token: punc ({) [./~/react-router-dom/umd/react-router-dom.production.min.js:11,866]

ubuntu: 22.04
node version: v16.14.0 (snap)
npm: 8.3.1

**Comment by cihatertem** [comment]
I solve my problem with "npm audit fix --force" but I think this is not best practice.
I ran this command two times:

-  At first time it upgraded "react-scripts": "0.9.5" to "react-scripts": "5.0.1" and I tried re-build but this time it gave me another error message.
- At second it downgraded "react-scripts": "5.0.1" to react-scripts": "2.1.3" and build process succeed.  Very very buggy and annoying process... :/
